src/Sarif.Driver/OrderedFileSpecifier.cs

```diff
@@ -47,11 +47,6 @@ private IEnumerable<IEnumeratedArtifact> EnumeratedArtifacts()
         {
             string normalizedSpecifier = Environment.ExpandEnvironmentVariables(this.specifier);
 
-            if (Uri.TryCreate(this.specifier, UriKind.RelativeOrAbsolute, out Uri uri))
```

Comment:
This isn't setting this.Uri.  It's creating a temp URI (which will be URI-encoded) so we can subsequently call GetFilePath on it and update the normalized specifier.  Uri should still be created as normal. (Wherever that happens.)

Reply:
this path normalization had impact on the filter and directory value we realized after this. now that normalization isn't occurring.

it's basically like eliminating a Path.GetFullPath call I suspect. i.e., constructs like `.\` and `..\..\` aren't resolved.

it's always dangerous to change this kind of behavior.

this code path is part of the core problem. the behavior here will differ depending on whether URI believes it has a file URI or not. that means for any relative paths, it will assume we have a web URI not a file URI. this is probably what's leading to encoding confusion